File: src/main/java/com/esprit/models/Equipement.java
package com.esprit.models;
import com.esprit.services.CategorieService;

public class Equipement {
    private int id;
    private String nom;
    private String description;
    private double prix;
    private int quantiteStock;
    private int idCategorie;
    private String image;// Référence à la catégorie par son ID

    // Constructeur avec ID
    public Equipement(int id, String nom, String description, double prix, int quantiteStock, int idCategorie, String image) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.prix = prix;
        this.quantiteStock = quantiteStock;
        this.idCategorie = idCategorie;
        this.image = image;
    }

    // Constructeur sans ID (utilisé avant insertion en BDD)
    public Equipement(String nom, String description, double prix, int quantiteStock, int idCategorie,String image) {
        this.nom = nom;
        this.description = description;
        this.prix = prix;
        this.quantiteStock = quantiteStock;
        this.idCategorie = idCategorie;
        this.image = image;
    }
    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image =image;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public int getQuantiteStock() {
        return quantiteStock;
    }

    public void setQuantiteStock(int quantiteStock) {
        this.quantiteStock = quantiteStock;
    }

    public int getIdCategorie() {
        return idCategorie;
    }

    public void setIdCategorie(int idCategorie) {
        this.idCategorie = idCategorie;
    }

    @Override
    public String toString() {
        return "Equipement{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", description='" + description + '\'' +
                ", prix=" + prix +
                ", quantiteStock=" + quantiteStock +
                ", idCategorie=" + idCategorie +
                '}';
    }
    public String getCategorieNom() {
        return CategorieService.getCategorieNomById(this.idCategorie);
    }

}

File: src/Entity/Equipement.php
<?php

namespace App\Entity;

use App\Repository\EquipementRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: EquipementRepository::class)]

class Equipement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(type: 'string', length: 255)]
    #[Assert\NotBlank(message: "Le nom ne peut pas être vide")]
    #[Assert\Length(
        min: 2,
        max: 255,
        minMessage: "Le nom doit contenir au moins {{ limit }} caractères",
        maxMessage: "Le nom ne peut pas dépasser {{ limit }} caractères"
    )]
    #[Assert\Regex(
        pattern: "/^[a-zA-ZÀ-ÿ\s\-']+$/u",
        message: "Le nom contient des caractères non autorisés"
    )]
    private ?string $nom = null;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    #[Assert\NotBlank(message: "Le description ne peut pas être vide")]
    #[Assert\Length(
        max: 255,
        maxMessage: "La description ne peut pas dépasser {{ limit }} caractères"
    )]
    private ?string $description=null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
#[Assert\NotBlank(message: "Le prix ne peut pas être vide")]
#[Assert\Positive(message: "Le prix doit être positif")]
#[Assert\LessThan(value: 1000000, message: "Le prix ne peut pas dépasser 1 000 000")]
private ?string $prix ; 
#[ORM\Column(type: 'integer')]
#[Assert\NotBlank(message: "La quantité ne peut pas être vide")]
#[Assert\PositiveOrZero(message: "La quantité doit être positive ou nulle")]
private ?int $quantiteStock = null;

    #[ORM\ManyToOne(targetEntity: CategorieEquipement::class)]
    #[ORM\JoinColumn(name: 'categorie_id', onDelete: 'SET NULL')]
    #[Assert\NotNull(message: "Veuillez sélectionner une catégorie")]
    private ?CategorieEquipement $categorie = null;
    #[ORM\Column(type: 'string', length: 255, nullable: true)]
  
    private ?string $image = null;








    // Getters et Setters

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom ? trim($nom) : null;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description ? trim($description) : null;;
        return $this;
    }

    public function getPrix(): string
    {
        return $this->prix;
    }

    public function setPrix(?string $prix): self
    {
     
        if ($prix === null || $prix === '') {
            $this->prix = null;
        } else {
            $prix = str_replace(',', '.', $prix);
            $this->prix = number_format((float)$prix, 2, '.', '');
        }
        return $this;
    }

    public function getQuantiteStock(): int
    {
        return $this->quantiteStock;
    }

    public function setQuantiteStock(?int $quantiteStock): self
    {
        $this->quantiteStock = $quantiteStock ?? 0; // Convertit null en 0
        return $this;
    }

    public function getCategorie(): ?CategorieEquipement
    {
        return $this->categorie;
    }

    public function setCategorie(?CategorieEquipement $categorie): static
    {
        $this->categorie = $categorie;
        return $this;
    }
    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }


    
}